File: ruby/easy/best_time_to_buy_and_sell_stock_ii.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    max_profit = 0
    for i in 1..prices.length-1
        profit = prices[i] - prices[i-1]
        max_profit += profit if profit > 0
    end
    return max_profit
end

File: javascript/easy/bestTimeToBuyAndSellStockII.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0, i = 0, j = 1;
    while(j < prices.length) {
        let profit = prices[j] - prices[i];
        if(profit > 0) {
            maxProfit += profit;
        }
        i++; j++;
    }
    return maxProfit;
};
